### DISCUSSION TITLE
Filter Event Logs from Contract in @wagmi/core

### DISCUSSION THREAD

**Discussion by saintmaxi** [wevm/viem#717] [Question]
Wondering if wagmi (specicially @wagmi/core) offers any wrapper around viem's `createContractEventFilter` (link here: https://viem.sh/docs/contract/createContractEventFilter.html#createcontracteventfilter). Trying to pull filtered events emitted from a contract (including additional filtering on indexed parameters). Is anyone doing this currently that could provide an example, whether that's with wagmi somehow or directly with viem?

Looks something like this in ethers:

```js
const contract = new ethers.Contract(address, abi);
const filter = contract.filters.FilterName(arg);

// events will contain an array of filtered events retrieved
const events = await contract.queryFilter(filter);
```

**Comment by xdrkush** [comment]
I have same question

**Comment by ECBSJ** [comment]
I also have the same question because I've also been having issues with `publicClient.createContractEventFilter` and `contractInstance.createEventFilter`. None of them are working for me.
